src/vm/perfmap.cpp

```diff
@@ -173,7 +179,7 @@ void PerfMap::LogMethod(MethodDesc * pMethod, PCODE pCode, size_t codeSize)
         // Build the map file line.
         StackScratchBuffer scratch;
         SString line;
-        line.Printf("%p %x %s\n", pCode, codeSize, fullMethodSignature.GetANSI(scratch));
+        line.Printf(FMT_TARGET_ADDR " %x %s\n", pCode, codeSize, fullMethodSignature.GetANSI(scratch));
```

Comment:
These are 32-bit offsets in crossgen generated map files, not actual addresses. It took me a while to figure this out.

@brianrob Would it make sense to always print these as 32-bit numbers in crossgen generated map files?

Reply:
pCode can either be an offset in the case where a native image perfmap is being built or an actual address in the case where we're building a perfmap for a live process.  In the latter, case PerfMap::LogMethod is called by PerfMap::LogJITCompiledMethod which takes the actual code address via pCode.  So, this does need to support writing the full width of a pointer.

Reply:
Ah.  As long as we can guarantee that the value will fit into 32-bits then it should be fine.  I'm not 100% sure about this as the values that we calculate this from are of type TADDR.  If you know it to be true that the values fit into 32-bits then that should be fine for the crossgen case.  I could also imagine us changing the format string (if possible) to just not write the full width of the value with prepended zeros and instead just write the minimal number of characters.  Then we get the savings on the JITed case too.